sormas-ui/src/main/java/de/symeda/sormas/ui/utils/CheckBoxTreeUpdated.java

```diff
@@ -0,0 +1,240 @@
+		return enumToggles;
+	}
+
+	@Override
+	protected Component initContent() {
+
+		if (enumType != null) {
+			if (groupProvider == null) {
+				checkBoxTreeLayout = new VerticalLayout();
+				checkBoxTreeLayout.setMargin(false);
+				checkBoxTreeLayout.setSpacing(false);
+				checkBoxTreeLayout.setWidth(100, Sizeable.Unit.PERCENTAGE);
+				addCheckBoxes();
+				final MarginInfo marginInfo = new MarginInfo(false, false, true, false);
+				checkBoxTreeLayout.setMargin(marginInfo);
+			} else {
+				checkBoxTreeLayoutForGroups = new HorizontalLayout();
+				checkBoxTreeLayoutForGroups.setMargin(false);
+				checkBoxTreeLayoutForGroups.setWidth(100, Unit.PERCENTAGE);
+				addGroups();
+			}
+
+			settingToggles = true;
+			setToggleValues(getValue());
+			settingToggles = false;
+		}
+
+		addValueChangeListener(valueChangeEvent -> {
+			Map<ENUM, Boolean> value = (Map<ENUM, Boolean>) valueChangeEvent.getProperty().getValue();
+			final Map<ENUM, Boolean> nullSafeValue = value == null ? new HashMap<>() : value;
+			settingToggles = true;
+			enumToggles.forEach(((anEnum, checkBox) -> {
+				final Boolean orDefault = nullSafeValue.getOrDefault(anEnum, false);
+				if (!Objects.equals(checkBox.getValue(), orDefault)) {
+					checkBox.setValue(orDefault);
+				}
+			}));
+			settingToggles = false;
+		});
+
+		if (checkBoxTreeLayout != null) {
+			return checkBoxTreeLayout;
+		}
+
+		return checkBoxTreeLayoutForGroups;
+	}
+
+	private void addGroups() {
+		final ENUM[] enumElements = enumType.getEnumConstants();
+
+		Map<Enum, List<ENUM>> enumGroupMap = new HashMap<>();
+
+		for (ENUM enumElement : enumElements) {
+			final Enum groupElement = groupProvider.apply(enumElement);
+			if (enumGroupMap.containsKey(groupElement)) {
+				enumGroupMap.get(groupElement).add(enumElement);
+			} else {
+				enumGroupMap.put(groupElement, new ArrayList<>());
+				enumGroupMap.get(groupElement).add(enumElement);
+			}
+		}
+
+		List<VerticalLayout> columnList = new ArrayList<>();
+
+		for (int i = 0; i < columns; i++) {
+			final VerticalLayout e = new VerticalLayout();
+			e.setMargin(false);
+			e.setSpacing(false);
+			columnList.add(e);
+		}
+
+		AtomicInteger currentColumn = new AtomicInteger();
+		currentColumn.set(0);
+		enumGroupMap.entrySet().stream().forEach(enumListEntry -> {
+			Label heading = new Label(enumListEntry.getKey().toString());
+			CssStyles.style(heading, CssStyles.H4);
+			columnList.get(currentColumn.get()).addComponent(heading);
+
+			enumListEntry.getValue().stream().forEach(anEnum -> {
+				CheckBox checkbox = createCheckbox(0, anEnum, null);
+				enumToggles.put(anEnum, checkbox);
+				columnList.get(currentColumn.get()).addComponent(checkbox);
+			});
+
+			if (currentColumn.get() == columns - 1) {
+				currentColumn.set(0);
+			} else {
+				currentColumn.getAndIncrement();
+			}
+		});
+
+		for (int i = 0; i < columns; i++) {
+			checkBoxTreeLayoutForGroups.addComponent(columnList.get(i));
```

Comment:
columns can be added right after creation, there is no neeed for another cycle for adding them to tha layout

Reply:
moved